### DISCUSSION TITLE
Can’t refresh data table with reactive rows

### DISCUSSION THREAD

**Discussion by myangga** [carbon-design-system/carbon-components-svelte#1713] [Q&A]
I’m not able to refresh data table component when source of rows changed. Is there any render action somehow? Or do I need to destroy the table and build new one?

**Comment by metonym** [comment]
Could you provide a minimal repro illustrating the issue?

**Reply by myangga** [reply]
Like pretty often issue between chair and computer... 😫 I did a Repl to reproduce issue but works! So issue is in my app code. Many thanks for your quick support. I publish my Repl anyway in case other people face issue: https://replit.com/@myangga/CDS-Datatable?v=1 . Thanks @metonym

**Reply by jackdarker** [reply]
Run into the same problem too, and I guess many other beginners will face this too with other components.
Thanks for the replit. This is basically the solution:
Instead of `let rows=[]` use `$: rows=[]` to register rows for triggering reevaluation for rows.
and wherever you modify rows you have to make sure to use ASSIGN because only assign triggers update:
`rows=[...rows, newitem]`  //just rows.push(newitem) will not work!.
